src/freenet/support/io/Fallocate.java

```diff
@@ -111,4 +118,28 @@ private static int getDescriptor(FileDescriptor descriptor) {
       throw new UnsupportedOperationException("unsupported FileDescriptor implementation", e);
     }
   }
+
+  public static void legacyFill(FileChannel fc, long newLength, long offset, boolean starting) throws IOException {
+    MersenneTwister mt = new MersenneTwister();
+    byte[] b = new byte[4096];
+    ByteBuffer bb = ByteBuffer.wrap(b);
+    int x = 0;
+    while (offset < newLength) {
+      bb.rewind();
+      mt.nextBytes(b);
+      offset += fc.write(bb, offset);
+      if (offset % (1024 * 1024 * 1024L) == 0) {
+        mt = new MersenneTwister();
+        if (starting) {
+          WrapperManager.signalStarting( (int) MINUTES.toMillis(5));
```

Comment:
We can't; NodeStarter.start() needs to return (WrapperListener.start()) ... otherwise the wrapper kicks us... so setting starting to true (and pinging WrapperManager.signalStarting periodically) is mandatory until that's happened.